src/app/home/home.page.ts

```diff
@@ -1,12 +1,282 @@
+
+    {
+      name: "On Tannaka Recognition and Descent for Topoi",
+      given_by: "me",
+      at: "2022 CMS Summer Meeting",
+      url_at: "https://summer22.cms.math.ca/",
+      url_abs: "https://www2.cms.math.ca/Events/summer22/abs/pdf/dma-ms.pdf",
+      url_video: "",
+      url_slides: "",
+      year: 2022,
+      date: "June 4",
+      proj: "Tannaka-Galois",
+    },
+
+    {
+      name: "An invitation to double categories and double fibrations",
+      given_by: "me",
+      at: "Algebra, Logic and Topology Seminar, Centre For Mathematics, Univ. of Coimbra",
+      url_at: "http://cmuc.mat.uc.pt › ppresearchgroupseminars",
+      url_abs: "https://www.mathstat.dal.ca/~mszyld/abstract_coimbra_2022.pdf",
+      url_video: "",
+      url_slides: "",
+      year: 2022,
+      date: "May 24",
+      proj: "Fibrations for Double Categories",
+    },
+
+    {
+      name: "Double Fibrations (or functors are categories)",
+      given_by: "me",
+      at: "ATCAT Seminar",
+      url_at: "https://www.mathstat.dal.ca/~fu/atcat/",
+      url_abs: "https://www.mathstat.dal.ca/~fu/atcat/",
+      url_video: "",
+      url_slides: "",
+      year: 2022,
+      date: "March 1",
+      proj: "Fibrations for Double Categories",
+    },
+
+    {
+      name: "Categories for the Learning Mathematician",
+      given_by: "me",
+      at: "Dalhousie Honours Seminar",
+      url_at: "https://www.dal.ca/faculty/science/math-stats/news-events/colloquium.html",
+      url_abs: "https://www.mathstat.dal.ca/~mszyld/abstract_szyld_honours_seminar.pdf",
+      url_video: "",
+      url_slides: "",
+      year: 2021,
+      date: "October 20",
+      proj: "Outreach",
+    },
+
+    {
+      name: "Minimal Bicategories of Fractions and Colimits of Bicategories",
+      given_by: "me",
+      at: "ATCAT Seminar",
+      url_at: "https://www.mathstat.dal.ca/~mamy/atcat/",
+      url_abs: "https://www.mathstat.dal.ca/~mamy/atcat/",
+      url_video: "",
+      url_slides: "",
+      year: 2021,
+      date: "October 12",
+      proj: "Minimal Fractions",
+    },
+
+  ];
+
+  public projects: Project[] = [{ name: "Minimal Fractions", description: "Minimal Fractions description", items: []},
+  { name: "Outreach", description: "Outreach description", items: []},
+  { name: "Fibrations for Double Categories", description: "Fibrations for Double Categories description", items: []},
+  { name: "Tannaka-Galois", description: "Tannaka-Galois description", items: []},
+  { name: "Model Bicategories", description: "Model Bicategories description", items: []},
+  { name: "Sigma Limits", description: "Sigma Limits description", items: []}];
+
+
+  public currentList: PersonalItem[] = this.articles;
+
+  public added_stuff = false;
+
+  constructor( ) {
+  }
+
+  segmentHasChanged(value) {
+    this.currentList = value.detail.value;
+    
+    // The first time value changes, create the list of items of each project:
+    if (this.added_stuff == false) {

```

Comment:
I would not do this in this class, rather make a helper service to load the data.

Reply:
https://angular.io/guide/architecture-services